RIOT-OS/RIOT
# cpu/esp32: Add openocd programming support

### Contribution description



Configures esp32 boards with `esp_jtag` support for flashing and debugging using openocd. 
Just like esptool it writes a bootloader, a partition table and the app when flashing.

### Testing procedure



Verify that esp8266 builds still work.

https://github.com/espressif/openocd-esp32 is needed (instead of normal openocd) to flash and debug the esp32.

I flashed my board using the following setup from inside an app directory:
```
$ echo $BOARD
esp32-ethernet-kit-v1_2
$ echo $OPENOCD
~/esp/openocd-esp32/src/openocd -s ~/esp/openocd-esp32/tcl
$ PROGRAMMER=openocd USEMODULE+=esp_jtag make -j flash
```

### Issues/PRs references



None
